test/curve_fit.jl

```diff
@@ -1,6 +1,6 @@
 let
     # fitting noisy data to an exponential model
-    model(x, p) = p[1]*exp(-x.*p[2])
+    model(x, p) = @compat p[1] .* exp.((-).(x) .* p[2])
```

Comment:
Yes, but it does it at the expense of readability. We've made various choices in LsqFit to sacrifice a small amount of performance in the name of readability. In this case, I find `(-).(x)` to be extraordinarily obtuse. So, please... let's use the easier to read version.